parallel.go

```diff
@@ -329,3 +331,289 @@ func ParAnd(parallelism int, bitmaps ...*Bitmap) *Bitmap {
 
 	return bitmap
 }
+
+// ParOr computes the union (OR) of all provided bitmaps in parallel,
+// where the parameter "parallelism" determines how many workers are to be used
+// (if it is set to 0, a default number of workers is chosen)
+func ParOr(parallelism int, bitmaps ...*Bitmap) *Bitmap {
+	var lKey uint16 = MaxUint16
+	var hKey uint16 = 0
+
+	bitmapsFiltered := bitmaps[:0]
+	for _, b := range bitmaps {
+		if !b.IsEmpty() {
+			bitmapsFiltered = append(bitmapsFiltered, b)
+		}
+	}
+	bitmaps = bitmapsFiltered
+
+	for _, b := range bitmaps {
+		lKey = minOfUint16(lKey, b.highlowcontainer.keys[0])
+		hKey = maxOfUint16(hKey, b.highlowcontainer.keys[b.highlowcontainer.size()-1])
+	}
+
+	if lKey == MaxUint16 && hKey == 0 {
+		return New()
+	} else if len(bitmaps) == 1 {
+		return bitmaps[0]
+	}
+
+	keyRange := hKey - lKey + 1
+	if keyRange == 1 {
+		// revert to FastOr. Since the key range is 0
+		// no container-level aggregation parallelism is achievable
+		return FastOr(bitmaps...)
+	}
+
+	if parallelism == 0 {
+		parallelism = defaultWorkerCount
+	}
+
+	var chunkSize int
+	var chunkCount int
+	if parallelism*4 > int(keyRange) {
+		chunkSize = 1
+		chunkCount = int(keyRange)
+	} else {
+		chunkCount = parallelism * 4
+		if int(keyRange)%chunkCount > 0 {
+			chunkSize = (int(keyRange) / chunkCount) + 1
+		} else {
+			chunkSize = int(keyRange) / chunkCount
+		}
+	}
+	//fmt.Printf("cc %d cs %d r %d d %d\n", chunkCount, chunkSize, int(keyRange)%chunkSize, chunkCount*chunkSize-int(keyRange))
```

Comment:
Do we need this commented out line?

Reply:
I left it for debugging issues, but you're right. We can remove it now.